app/plugins/eu.esdihumboldt.hale.app.transform/src/eu/esdihumboldt/hale/app/transform/ExecApplication.groovy

```diff
@@ -227,23 +229,18 @@ $baseCommand
 				break
 
 			case '-filter-on':
-			//<Type>:CQL:<EXPRESSION>
-				if(value.findIndexValues {it = ':'}.size() >=2 ){
-					int typeNameIndex = value.indexOf(':')
-					if(typeNameIndex>0){
-						String typeName = value.take(typeNameIndex)
-						String expression = value.substring(typeNameIndex+1)
-						executionContext.filters.addTypeFilter(typeName, expression)
-					}
-				}else {
+				int typeNameIndex = value.indexOf(':')
```

Comment:
Thinking further about it we have a problem here with the colon. It's not a rare case that there is a colon used in a namespace. In the example I think the fully fledged QName would be represented like this:

`{eu:esdihumboldt:hale:example}RiverType`

That means that this approach to separating the type name from the other arguments does not work.